app/controllers/discover.js

```diff
@@ -171,6 +171,15 @@ export default Ember.Controller.extend({
             this.set('sortByOptions', copy);
             this.set('page', 1);
             this.loadPage();
+        },
+
+        selectSubjectFilter(subject) {
+            let match = this.get('activeFilters').filter(function(item) {
+                return item.indexOf(subject.text) !== -1;
+            });
+            if (!match.length) {
+                this.get('activeFilters').pushObject('subject:' + subject.text);
```

Comment:
Why is there a need to prefix it with "subject:"? Is this because of previous understanding of how share would send subjects?